### PR TITLE
Enable StaticTransformBroadcaster in Intra-process enabled components

### PR DESCRIPTION
**Pull request by roncapat**
Follow up of [this](https://github.com/ros2/geometry2/pull/572#issuecomment-1361192871) comment made by @FranzAlbers (and PR #572 in general).

FYI  @nachovizzo(https://github.com/PRBonn/kiss-icp/pull/171) and @fixit-davide.

### CODE REVIEW

File: tf2_ros/include/tf2_ros/static_transform_broadcaster.h [hunk 1]
```diff
@@ -66,6 +66,14 @@ class StaticTransformBroadcaster
         rclcpp::QosPolicyKind::Depth,
         rclcpp::QosPolicyKind::History,
         rclcpp::QosPolicyKind::Reliability};
+      /*
+        This flag disables intra-process communication while subscribing to
+        /tf topic, when the TransformListener is constructed using an existing
```
Comment (clalancette) [inline tf2_ros/include/tf2_ros/static_transform_broadcaster.h]:
```suggestion
        /tf_static topic, when the TransformListener is constructed using an existing
```

File: tf2_ros/include/tf2_ros/static_transform_broadcaster.h [hunk 2]
```diff
@@ -87,6 +95,14 @@ class StaticTransformBroadcaster
...
```
Comment (clalancette) [inline tf2_ros/include/tf2_ros/static_transform_broadcaster.h]:
```suggestion
        /tf_static topic, when the TransformListener is constructed using an existing
```

**Review by clalancette — CHANGES_REQUESTED** [review]
Two small changes, then I'll run CI on it.

**Comment by roncapat** [comment]
I also added a small test as I did for the TransformListener :)

**Review by clalancette — APPROVED** [review]
Looks good, thanks for the addition of the test!

**Comment by roncapat** [comment]
@clalancette wait wait, I missed some fixes to the comments
sorry!

EDIT: done, just need to see if uncrustify is happy with different newlines in the comment blocks. Sorry again for that :)

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19014/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13537/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/19766/)

**Comment by roncapat** [comment]
Thanks a lot! Can we backport to iron and humble?

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport iron humble

**Pull request merged**